src/install/install.zig

```diff
@@ -5086,85 +5086,77 @@ pub const PackageManager = struct {
-                            @tagName(op),
-                        });
-                    } else {
-                        Output.prettyErrorln("<r><red>error<r><d>:<r> bun {s} http://url is not implemented yet.", .{
-                            @tagName(op),
-                        });
-                    }
-
+                const placeholder = String.from("@@@");
+                var version = Dependency.parseWithOptionalTag(
+                    allocator,
+                    placeholder,
+                    value,
+                    null,
+                    &SlicedString.init(value, value),
+                    log,
+                ) orelse switch (op) {
+                    .link, .unlink => null,
+                    else => brk: {
+                        value = std.fmt.allocPrint(allocator, "npm:{s}", .{value}) catch unreachable;
+                        break :brk Dependency.parseWithOptionalTag(
+                            allocator,
+                            placeholder,
+                            value,
+                            null,
+                            &SlicedString.init(value, value),
+                            log,
+                        );
+                    },
+                } orelse {
+                    Output.prettyErrorln("<r><red>error<r><d>:<r> unrecognised dependency format: {s}", .{
+                        positional,
+                    });
                     Global.exit(1);
+                };
+                switch (version.tag) {
+                    .dist_tag => if (version.value.dist_tag.name.eql(placeholder, value, value)) {
+                        value = std.fmt.allocPrint(allocator, "npm:{s}", .{value}) catch unreachable;
+                        version = Dependency.parseWithOptionalTag(
+                            allocator,
+                            placeholder,
+                            value,
+                            null,
+                            &SlicedString.init(value, value),
+                            log,
+                        ) orelse {
+                            Output.prettyErrorln("<r><red>error<r><d>:<r> unrecognised dependency format: {s}", .{
+                                positional,
+                            });
+                            Global.exit(1);
+                        };
+                    },
+                    else => {},
                 }
 
-                request.name = std.mem.trim(u8, request.name, "\n\r\t");
-                if (request.name.len == 0) continue;
-
+                var request = UpdateRequest{
+                    .name = switch (version.tag) {
+                        .dist_tag => version.value.dist_tag.name,
+                        .github => version.value.github.repo,
+                        .npm => version.value.npm.name,
+                        .symlink => version.value.symlink,
+                        else => version.literal,
+                    }.slice(value),
```

Comment:
... as in right here, it is being converted from `String` (stack) to `string` (heap?)

Reply:
if the name/version is <= 8 characters, the pointer will be to the inline `String` instead of the heap

Reply:
Done in https://github.com/oven-sh/bun/pull/1895/commits/cd3aa41a203b0876ee9ab9423856d8833af18140